pkg/kubectl/resize.go

```diff
@@ -70,11 +93,28 @@ type ReplicationControllerResizer struct {
 	client.Interface
 }
 
+// ResizeCondition is a closure around Resize that facilitates retries via util.wait
+func ResizeCondition(r Resizer, precondition *ResizePrecondition, namespace, name string, count uint) wait.ConditionFunc {
+	return func() (bool, error) {
+		_, err := r.Resize(namespace, name, precondition, count)
+		switch err.(type) {
+		case nil:
+			return true, nil
+		case ControllerResizeError:
+			e := err.(ControllerResizeError)
+			if e.FailureType == UpdateFailure {
+				return false, nil
```

Comment:
Ideas for another PR: It would be good to error out if you get something other than a conflict error. Like, if you're getting this error because validation failed trying again won't change that.